File: Data Structures & Algorithms/sliding-window-maximum/submission-9.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {

        int[] result = new int[nums.length - k + 1];
        Deque<Integer> queue = new ArrayDeque<>();
        int left = 0, right = 0;

        while(right <= nums.length - 1) {
            while(!queue.isEmpty() && nums[queue.getLast()] < nums[right]) {
                queue.removeLast();
            }
            queue.offer(right);
            if(left > queue.getFirst()) {
                queue.removeFirst();
            }
            if(right + 1 >= k) {
                result[left] = nums[queue.getFirst()];
                left++;
            }
            right++;
        }
        return result;
    }
}

File: Data Structures & Algorithms/majority-element/submission-9.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = dict()

        for num in nums:
            count[num] = 1 + count.get(num, 0)
        
        for num in nums:
            if count[num] > len(nums) / 2:
                return num
        